src/krb5/_keyblock_mit.pyx

```diff
@@ -0,0 +1,69 @@
+# Copyright: (c) 2022 Jordan Borean (@jborean93) <jborean93@gmail.com>
+# MIT License (see LICENSE or https://opensource.org/licenses/MIT)
+
+from krb5._exceptions import Krb5Error
+from krb5._keyblock import init_keyblock
+
+from krb5._context cimport Context
+from krb5._keyblock cimport KeyBlock
+from krb5._krb5_types cimport *
+
+
+cdef extern from "python_krb5.h":
+    krb5_error_code krb5_c_string_to_key(
+        krb5_context context,
+        krb5_enctype enctype,
+        const krb5_data *string,
+        const krb5_data *salt,
+        krb5_keyblock *key,
+    ) nogil
+
+    krb5_error_code krb5_c_string_to_key_with_params(
+        krb5_context context,
+        krb5_enctype enctype,
+        const krb5_data *string,
+        const krb5_data *salt,
+        const krb5_data *params,
+        krb5_keyblock *key,
+    ) nogil
+
+
+def c_string_to_key(
+    Context context not None,
+    krb5_enctype enctype,
+    const unsigned char[:] string,
+    const unsigned char[:] salt,
+    const unsigned char[:] s2kparams = None,
+) -> KeyBlock:
+    cdef krb5_error_code err = 0
+    cdef size_t length = 0
+
+    cdef KeyBlock kb = init_keyblock(context, enctype, None)
+
+    cdef krb5_data string_raw
+    if len(string) == 0:
+        pykrb5_set_krb5_data(&string_raw, 0, "")
```

Comment:
Should this be `""` or `NULL`?

Reply:
I think it should not matter, it simply has to be a pointer to 0 bytes of data, so both the NULL pointer or a pointer to an empty string (or even some other string) should work. The only reason `len(string) == 0` is a separate branch is because in the else branch `<char *>&string[0]` will fail if `string` is empty.